### PR TITLE
Jan Lehigh 2403 Back End

### PR DESCRIPTION
**Pull request by JCL461437**

### CODE REVIEW

**Comment by JCL461437** [comment]
I had to re clone my repository. I am not sure if that deleted the history of my old commits, but if there seems to be a lack of them that may be why. Thanks for looking!

File: spec/event_spec.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            expect(@event).to be_an_instance_of(Event)
+        end
+        
+        it 'can return event name' do
+            expect(@event.name).to eq("South Pearl Street Farmers Market")
+        end
+
+        it 'can return food trucks at event' do
+            expect(@event.food_trucks).to eq([])
+        end
```
Comment (KatBrandt) [inline spec/event_spec.rb:32]: These tests above look good.

File: spec/event_spec.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            @food_truck3.stock(@item1, 65)  
+
+            @event.add_food_truck(@food_truck1) 
+            expect(@event.food_trucks).to eq([@food_truck1])
+
+            @event.add_food_truck(@food_truck2) 
+            expect(@event.food_trucks).to eq([@food_truck1, @food_truck2])
+
+            @event.add_food_truck(@food_truck3) 
+            expect(@event.food_trucks).to eq([@food_truck1, @food_truck2, @food_truck3])
```
Comment (KatBrandt) [inline spec/event_spec.rb:52]: Really think about what you actually need in your setup for a test. Here, all we're trying to make sure of is that we can add trucks, so we really doing need to stock the items before doing that.

File: spec/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            @food_truck1.stock(@item1, 35) 
+            @food_truck1.stock(@item2, 7)
+            @food_truck2.stock(@item4, 50)  
+            @food_truck2.stock(@item3, 25)
+            @food_truck3.stock(@item1, 65)  
+
+            @event.add_food_truck(@food_truck1) 
+            @event.add_food_truck(@food_truck2)
+            @event.add_food_truck(@food_truck3) 
+            expect(@event.food_trucks).to eq([@food_truck1, @food_truck2, @food_truck3])
```
Comment (KatBrandt) [inline spec/event_spec.rb:68]: Don't need this assertion line since we already tested this in the test before.

File: spec/event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+            expect(@event.food_truck_names).to eq(["Rocky Mountain Pies", "Ba-Nom-a-Nom", "Palisade Peach Shack"])
```
Comment (KatBrandt) [inline spec/event_spec.rb:87]: Same here, lines 85 and 87 are not necessary

File: spec/event_spec.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+            expect(@event.food_trucks_that_sell(@item1)).to eq([@food_truck1, @food_truck3])
+            expect(@event.food_trucks_that_sell(@item4)).to eq([@food_truck2])
+        end
```
Comment (KatBrandt) [inline spec/event_spec.rb:91]: Nice job testing two different trucks to make sure you're method is working for different objects.

File: spec/food_truck_spec.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        it 'can check stock for an item' do
+            expect(@food_truck1.check_stock(@item1)).to eq(0)
+        end
+    end
+
+    describe '#stock' do
+        it 'can stock an item' do
+            @food_truck1.stock(@item1, 30)
+            expect(@food_truck1.inventory).to eq({@item1 => 30})
+        end
```
Comment (KatBrandt) [inline spec/food_truck_spec.rb:43]: We want to start getting more thorough in our tests. You checked that you can stock one item, that's great, but then after line 42 I would stock item1 again, then make sure the inventory hash looked correct, then I would have stocked item2 (any other item) and tested one last time that my hash looked correct. The more thorough we arein our tests, the more we can have confidence in our methods.

File: spec/food_truck_spec.rb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        it "can calculate potential total revenue" do
+            @food_truck1.stock(@item1, 35) 
+            @food_truck1.stock(@item2, 7)
+            @food_truck2.stock(@item4, 50)  
+            @food_truck2.stock(@item3, 25)
+            @food_truck3.stock(@item1, 65)  
+
+            expect(@food_truck1.potential_revenue). to eq (148.75)
+            expect(@food_truck2.potential_revenue). to eq (345.00)
+            expect(@food_truck3.potential_revenue). to eq (243.75)
```
Comment (KatBrandt) [inline spec/food_truck_spec.rb:63]: Like this one is good, very thorough test.

**Review by KatBrandt — COMMENTED** [review]
Great start here Jan! I recommend continuing to work through this IC, use student support and mentors to really work through all of iteration 3.

**Pull request closed without merging**